client/my-sites/controller.js

```diff
@@ -183,6 +185,18 @@ module.exports = {
 				// if sites have loaded, but siteID is invalid, redirect to allSitesPath
 				if ( sites.select( siteID ) ) {
 					onSelectedSiteAvailable();
+				} else if ( ( currentUser.visible_site_count !== sites.getVisible().length ) ) {
+					ReactDom.render( (
+						<Main>
+							<JetpackManageErrorPage template="waitForJetpackSite" />
+						</Main>
+					), document.getElementById( 'primary' ) );
+
+					setTimeout( () => {
```

Comment:
Basically the logic in that place would be "once sites resolve, if site length is still not the same as the site_count attribute of the user, run another fetch and wait for the change event again".